core/src/main/java/org/jivesoftware/sparkimpl/profile/AvatarPanel.java

```diff
@@ -291,7 +294,44 @@ public void setParentDialog(Dialog dialog) {
         this.dlg = dialog;
     }
 
+    /*
+     * Crop the given image file to the target dimensions 
+     * without changing aspect ratio.
+     * Returns modified image as BufferedImage.  
+     */
+    private BufferedImage cropImageFile(File selectedFile) {
+
+        int targetImageWidth = 64;
+        int targetImageHeight = 64;
+
+        BufferedImage croppedImage = null;
+        try {
+            BufferedImage avatarImageBuffered = ImageIO.read(selectedFile);
+
+            int currentImageWidth = avatarImageBuffered.getWidth(null);
+            int currentImageHeight = avatarImageBuffered.getHeight(null);
+            
+            if (currentImageHeight == targetImageHeight && currentImageWidth == targetImageHeight) {
+                //no need to crop
+                croppedImage = avatarImageBuffered;
+            } else {
+                double targetImageRatio = (double)targetImageWidth / (double)targetImageHeight;
+
+                if (currentImageWidth < targetImageRatio * currentImageHeight) {
+                    currentImageHeight = new Double((double)currentImageWidth / (double)targetImageRatio).intValue();
+                } else {
+                    currentImageWidth = new Double(((double)currentImageHeight * targetImageRatio)).intValue();
+                }
+                //crop image in the center
+                croppedImage = Thumbnails.of(avatarImageBuffered).sourceRegion(Positions.CENTER, currentImageWidth, currentImageHeight).
+                        size(targetImageWidth,targetImageWidth).asBufferedImage();
```

Comment:
``size(targetImageWidth,targetImageWidth)`` probably needs to be ``size(targetImageWidth,targetImageHeight)``, even though they're both 64.